tests/loginTest.js

```diff
@@ -49,63 +44,72 @@ describe('Main: Currently testing Login and initLibrary,', function () {
     }
   })
 
-  it('checking method : login with wrong config.username and config.password', function (done) {
-    oc = new OwnCloud({
-      baseUrl: config.owncloudURL,
-      auth: {
-        basic: {
-          username: nonExistentUser,
-          password: 'config.password' + timeRightNow
+  it('checking method : login with wrong username and password', async function () {
+    const provider = createProvider()
+    await capabilitiesGETRequestInvalidAuth(provider, nonExistentUser, 'config.password' + timeRightNow)
+
+    return provider.executeTest(async () => {
+      oc = new OwnCloud({
+        baseUrl: config.owncloudURL,
+        auth: {
+          basic: {
+            username: nonExistentUser,
+            password: 'config.password' + timeRightNow
+          }
         }
-      }
-    })
+      })
 
-    oc.login().then(() => {
-      fail()
-      done()
-    }).catch(error => {
-      expect(error).toMatch('Unauthorized')
-      done()
+      return oc.login().then(() => {
+        fail()
+      }).catch(error => {
+        expect(error).toMatch('Unauthorized')
+      })
     })
   })
 
-  it('checking method : login with correct config.username only', function (done) {
-    oc = new OwnCloud({
-      baseUrl: config.owncloudURL,
-      auth: {
-        basic: {
-          username: config.username,
-          password: 'config.password' + timeRightNow
+  it('checking method : login with correct username only', async function () {
+    const provider = createProvider()
+    await capabilitiesGETRequestInvalidAuth(provider, config.username, 'config.password' + timeRightNow)
+
+    return provider.executeTest(async () => {
+      oc = new OwnCloud({
+        baseUrl: config.owncloudURL,
+        auth: {
+          basic: {
+            username: config.username,
+            password: 'config.password' + timeRightNow
+          }
         }
-      }
-    })
+      })
 
-    oc.login().then(() => {
-      fail()
-      done()
-    }).catch(error => {
-      expect(error).toMatch('Unauthorized')
-      done()
+      return oc.login().then(() => {
+        fail()
+      }).catch(error => {
+        expect(error).toMatch('Unauthorized')
+      })
     })
   })
 
-  it('checking method : login with correct config.username and config.password', function (done) {
-    oc = new OwnCloud({
-      baseUrl: config.owncloudURL,
-      auth: {
-        basic: {
-          username: config.username,
-          password: config.password
+  it('checking method : login with correct config.username and config.password', async function () {
+    const provider = createProvider()
+    await capabilitiesGETRequestValidAuth(provider)
+    await GETRequestToCloudUserEndpoint(provider)
+    return provider.executeTest(async () => {
+      oc = new OwnCloud({
+        baseUrl: config.owncloudURL,
+        auth: {
+          basic: {
+            username: config.username,
+            password: config.password
+          }
```

Comment:
we can use `createOwncloud` from the Helper, but as it was like that before, it can be done in a separate PR